### PR TITLE
Fixed database tests

### PR DESCRIPTION
**Pull request by grzesiek2010**
#2849 ## What has been done to verify that this works as intended?
I just confirmed my changes fix the tests.

#### Why is this the best possible solution? Were any other approaches considered?
I just fixed the failing tests and improved them a little.

#### How does this change affect users? Describe intentional changes to behavior and behavior that could have accidentally been affected by code changes. In other words, what are the regression risks?
It doesn't require testing.

#### Do we need any specific form for testing your changes? If so, please attach one.
No.

#### Does this change require updates to documentation? If so, please file an issue [here]( https://github.com/opendatakit/docs/issues/new) and include the link below.
No.

#### Before submitting this PR, please make sure you have:
- [x] run `./gradlew checkAll` and confirmed all checks still pass OR confirm CircleCI build passes and run `./gradlew connectedDebugAndroidTest` locally.
- [x] verified that any code or assets from external sources are properly credited in comments and/or in the [about file](https://github.com/opendatakit/collect/blob/master/collect_app/src/main/assets/open_source_licenses.html).
- [x] verified that any new UI elements use theme colors. [UI Components Style guidelines](https://github.com/opendatakit/collect/blob/master/CONTRIBUTING.md#ui-components-style-guidelines)

### CODE REVIEW

File: collect_app/src/main/java/org/odk/collect/android/dto/Form.java [hunk 1]
```diff
@@ -237,4 +225,20 @@ public String getAutoDelete() {
     public String getLastDetectedFormVersionHash() {
         return lastDetectedFormVersionHash;
     }
+
+    @Override
+    public boolean equals(Object other) {
```
Comment (lognaturel) [inline collect_app/src/main/java/org/odk/collect/android/dto/Form.java:230]: For both `equals` and `hashCode`, I think you can just use `md5Hash`. It feels really odd to have a hash of a hash. And the display name and form ID are implicitly part of the hash, right?

File: collect_app/src/main/java/org/odk/collect/android/dto/Instance.java [hunk 2]
```diff
@@ -157,15 +149,27 @@ public Long getLastStatusChangeDate() {
...
         return deletedDate;
     }
 
     public Long getDatabaseId() {
         return databaseId;
     }
+
+    @Override
+    public boolean equals(Object other) {
+        return other == this || other instanceof Instance
```
Comment (lognaturel) [inline collect_app/src/main/java/org/odk/collect/android/dto/Instance.java:162]: How did you pick which subset of fields to include here? For example, why not all fields? Why display name and not version? Why not only instance file path or only instanceId?

**Comment by lognaturel** [comment]
The test changes look good. I think that at some point we should consider making this a UI-based test since it runs in Espresso anyway but fixing it as-is seems like a great start.

This adds `equals` and `hashCode` methods to the `Form` and `Instance` data transfer objects. Writing `equals` and `hashCode` methods is quite an art 🎨. 

I've made some suggestions inline. Additionally, either the classes or the methods should be final -- https://jqno.nl/equalsverifier/manual/final/. That site has a lot of other descriptions of issues with `equals` and `hashCode`. I think if those methods are going to be introduced, they should be done really carefully. Another option to consider is just comparing `md5hash` and `instanceId` in the tests so those methods don't need to be added now.

File: collect_app/src/main/java/org/odk/collect/android/dto/Form.java (hunk 1 above)
Comment (grzesiek2010) [reply collect_app/src/main/java/org/odk/collect/android/dto/Form.java:230]: Yeah md5Hash is enough. Fixed.

File: collect_app/src/main/java/org/odk/collect/android/dto/Instance.java (hunk 2 above)
Comment (grzesiek2010) [reply collect_app/src/main/java/org/odk/collect/android/dto/Instance.java:162]: We don't have instanceId column but yeah I can just use instance file path. FIxed.

**Comment by grzesiek2010** [comment]
> Additionally, either the classes or the methods should be final

Done.

**Review by lognaturel — APPROVED** [review]

**Pull request merged**